docs/source/schema-directives.md

```diff
@@ -1,179 +1,581 @@
 
-To start, let's grab the schema definition string from the `makeExecutableSchema` example [in the "Generating a schema" article](/tools/graphql-tools/generate-schema.html#example).
+type ExampleType {
+  newField: String
+  oldField: String @deprecated(reason: "Use `newField`.")
+}
+```
+
+Points to notice in this example:
+
+* The `@deprecated` directive _follows_ the field that it pertains to (`oldField`), even though its syntax might remind you of "decorators" in other languages, which usually appear on the line above.
+* Directives are often _declared_ in the schema, as in this example, though it's up to the GraphQL server to enforce the argument types (`reason: String`) and locations (`FIELD_DEFINITION | ENUM_VALUE`) of the declaration.
+* The `@deprecated(reason: ...)` syntax is legal even without the `directive @deprecated ...` declaration.
+* Whether a directive has been declared or not, the GraphQL server is responsible for giving it meaning, or else it will be ignored after parsing, almost as if it was a comment.
+
+While only a few directives are [required](http://facebook.github.io/graphql/draft/#sec--skip) by the specification, the formal syntax of the GraphQL query and schema languages allows arbitrary user-defined directives to appear as modifiers following almost any kind of type, field, or argument. If the server knows how to interpret them, these directive annotations can be a powerful tool for preventing repetition, specifying extra behavior, enforcing additional type or value restrictions, and enabling static analysis.
+
+This document focuses on directives that appear in GraphQL _schemas_ (as opposed to queries) written in [Schema Definition Language](https://github.com/facebook/graphql/pull/90), or SDL for short. Specifically, the following sections will explain how custom directives can be implemented and used to modify the structure and behavior of a GraphQL schema in ways that would not be possible using SDL syntax alone.
+
+The possible applications of directive syntax are numerous: enforcing access permissions, formatting date strings, auto-generating resolver functions for a particular backend API, marking strings for internationalization, synthesizing globally unique object identifiers, specifying caching behavior, skipping or including or deprecating fields, and just about anything else you can imagine.
+
+## Implementing schema directives
+
+Since the GraphQL specification does not discuss any specific implementation strategy for directives, it's up to each GraphQL server framework to expose an API for implementing new directives.
+
+If you're using Apollo Server, you are also likely to be using the [`graphql-tools`](https://github.com/apollographql/graphql-tools) npm package, which provides a convenient yet powerful tool for implementing directive syntax: the [`SchemaDirectiveVisitor`](https://github.com/apollographql/graphql-tools/blob/wip-schema-directives/src/schemaVisitor.ts) class.
```

Comment:
This page is in the `graphql-tools` docs, so referencing Apollo Server isn't really necessary in this way: https://deploy-preview-640--graphql-tools-docs.netlify.com/docs/graphql-tools/schema-directives.html

Reply:
That's amazing!